dastardcommander/trigger_blocker.py

```diff
@@ -0,0 +1,109 @@
+import json
+import os
+
+
+class TriggerBlocker:
+    """
+    Handle a more-or-less permanent list of channels that are considered "bad" and should be
+    blocked from triggering no matter what the user asks for. That is, any "all channels" settings
+    will not allow triggering of these channels.
+
+    Values are stored to and read from a config file (default: ~/.dastard/blocked_channels.json),
+    including a history of the last several lists.
+
+    Usage:
+    >>> tb = TriggerBlocker(tempfile.mkstemp(suffix=".json")[1])
+    >>> tb.block_channels(6,4,2)
+    >>> print(tb.blocked)
+    [2, 4, 6]
+    >>> tb.block_channels(8,6,[4,2],4,6,8)
+    >>> print(tb.blocked)
+    [2, 4, 6, 8]
+    >>> tb.unblock_channels(4)
+    >>> print(tb.blocked)
+    [2, 6, 8]
+    >>> tb.revert_prev()
+    >>> print(tb.blocked)
+    [2, 4, 6, 8]
+    >>> tb.clear()
+    >>> print(tb.blocked)
+    []
+    """
+
+    FIELDS_TO_MEMO = ("blocked", "blocked_history")
+    HISTORY_LENGTH = 10
+
+    def __init__(self, configfile=None):
+        if configfile is None:
+            configfile = "~/.dastard/blocked_channels.json"
+        self.config = os.path.expanduser(configfile)
+        self.blocked = []
+        self.blocked_history = []
+        self.read_config()
+
+    def read_config(self):
+        try:
+            with open(self.config, "r") as fp:
+                data = json.load(fp)
+                for field in self.FIELDS_TO_MEMO:
```

Comment:
instead of `FIELDS_TO_MEMO` I would use `dataclasses_json` and write a class that has everything you want to json serialize/derserialize, then `read_config` would return an instance of this class, so the self mutation of your object isn't baked into the reading process.

Reply:
I only halfway understand this comment. Unless you feel more strongly than it sounds like, I think I'll proceed this way.